artman/tasks/python_grpc_tasks.py

```diff
@@ -96,45 +100,49 @@ def _extract_base_dirs(self, proto_file):
                     return os.path.sep.join(pkg.split('.'))
             return ''
 
-    def _transform(self, pkg, sep, common_protos):
+    def _transform(self, pkg, sep, common_protos, api_name, api_version):
         """Transform to the appropriate proto package layout.
 
         Works with arbitrary separator (e.g., '/' for import statements,
         '.' for proto package statements, os.path.sep for filenames)
         """
+        # Split the proto path into a list, and move the `.proto` extension
+        # off to the side for now.
         if sep != '.' and pkg.endswith('.proto'):
-            dotted = pkg[:-6].replace(sep, '.')
+            dotted = pkg[:-6].replace(sep, '.').split('.')
             suffix = '.proto'
         else:
-            dotted = pkg.replace(sep, '.')
+            dotted = pkg.replace(sep, '.').split('.')
             suffix = ''
 
-        # Sanity check: Do not transform common protos.
```

Comment:
Does this mean we don't need to special case common protos any more?

Reply:
This is part of why I marked it `do not merge`; we need to talk about this aspect. The current logic will bypass any case where `api_name` and `api_version` are not present in the proto path. This does mean that common protos are no longer a special case.

However, there is another effect here, which is that transitive APIs would no longer get caught either. We had test cases for `google.service.v1` and `google.other_api.v2`, and it is no longer the case that the latter of the two would be transformed at all.

We need to discuss whether this is a case we actually needed, because if so, I just created a problem. :-)